### PR TITLE
Added observable to_future and to_stream

### PR DESCRIPTION
**Pull request by Neo-Ciber94**
This pull request implements the methods `to_future` and `to_stream`

# Implementation details

The implementation makes use of the `futures::channel::mpsc::unbounded` channel to send the values and poll them in the receiver.

For the future the output is of the type `Result<Result<T, E>, ObservableError` which fails when the observable is empty or emit more than 1 value.

### CODE REVIEW

**Comment by M-Adoo** [comment]
@Neo-Ciber94 thanks!  Really important operators to Interaction with `Future`， but   #215 will merge in this week, please base on it to implement `to_future` & `to_stream`.

**Comment by Neo-Ciber94** [comment]
@M-Adoo sure!

**Comment by Neo-Ciber94** [comment]
I made some changes after pulling the changes from the `refactor-traits` branch and the  test `ops::throttle::tests::smoke_for_throttle_time` is failing

File: src/observable.rs [hunk 1]
```diff
@@ -1621,6 +1623,22 @@ pub trait ObservableExt<Item, Err>: Sized {
   fn complete_status(self) -> (StatusOp<Self>, Arc<CompleteStatus>) {
     ops::complete_status::complete_status(self)
   }
+
+  #[inline]
```
Comment (M-Adoo) [inline src/observable.rs:1627]: This is the most important api for user,  some comments need. It's better to give an example give.

**Review by M-Adoo — COMMENTED** [review]
#215 is merged, please rebase the master and squash the commits.

File: src/observable.rs (hunk 1 above)
Comment (Neo-Ciber94) [reply src/observable.rs]: Documentation added

File: src/subject.rs [hunk 2]
```diff
@@ -278,6 +278,7 @@ where
   impl_observable_for_subject!(Subscriber);
 }
 
+// todo: support pass reference.
```
Comment (M-Adoo) [inline src/subject.rs:281]: unnecessary comment

**Review by M-Adoo — COMMENTED** [review]
Great! Thanks, only an unnecessary comment need remove, seems come from rebase conflict, and please update the `CHANGELOG.md`.

File: src/subject.rs (hunk 2 above)
Comment (Neo-Ciber94) [reply src/subject.rs]: Removed comment, and updated CHANGELOG

I added the `to_future` and `to_stream` in the **1.0.0-alpha.4** not sure if is unreleased

**Review by M-Adoo — APPROVED** [review]

**Comment by M-Adoo** [comment]
@Neo-Ciber94 the document example compile not pass.

**Comment by Neo-Ciber94** [comment]
> @Neo-Ciber94 the document example compile not pass.

fixed

**Review by M-Adoo — APPROVED** [review]

**Comment by Neo-Ciber94** [comment]
I need to move some tokio features to avoid conflicts with wasm

**Comment by Neo-Ciber94** [comment]
@M-Adoo I updated the Cargo.toml to exclude the dev-dependency tokio **rt-multi-thread** _(which is used in the doc tests of to_future/to_stream)_ from wasm

**Review by M-Adoo — APPROVED** [review]

**Pull request merged**